sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.ts

```diff
@@ -343,4 +354,144 @@ export class AssessmentSequentialProcessingService {
+          student: {
+            id: true,
+            birthDate: true,
+            sinValidation: {
+              sin: true,
+            },
+            user: {
+              lastName: true,
+            },
+          },
+          programYear: {
+            id: true,
+            startDate: true,
+          },
+        },
+        assessmentDate: true,
+      },
+      relations: {
+        application: {
+          student: { sinValidation: true, user: true },
+          programYear: true,
+        },
+      },
+      where: {
+        id: assessmentId,
+      },
+    });
+    if (!assessment) {
+      throw new Error(`Assessment if ${assessmentId} not found.`);
+    }
+    const lastName = assessment.application.student.user.lastName;
+    const sin = assessment.application.student.sinValidation.sin;
+    const birthDate = assessment.application.student.birthDate;
+    const programYearStartDate = assessment.application.programYear.startDate;
+    // Get the first assessment date for the current application. If there are multiple
+    // assessments for the current application, then set the first assessment date to
+    // the assessment date of the first assessment.
+    let firstAssessmentDate = getISODateOnlyString(assessment.assessmentDate);
+    const firstAssessment = await this.dataSource.manager
+      .getRepository(StudentAssessment)
+      .createQueryBuilder("assessment")
+      .select("assessment.assessmentDate")
+      .innerJoin("assessment.application", "application")
+      .where("assessment.id = :assessmentId", { assessmentId: assessment.id })
+      .where("application.applicationNumber = :applicationNumber", {
+        applicationNumber: assessment.application.applicationNumber,
+      })
+      .orderBy("assessment.assessmentDate")
+      .getOne();
+    if (firstAssessment) {
+      firstAssessmentDate = getISODateOnlyString(
+        firstAssessment.assessmentDate,
+      );
+    }
+    const sfasApplicationLegacyRepo =
+      sfasApplicationType === "sfasPTApplication"
+        ? this.sfasPartTimeApplicationsRepo
+        : this.sfasApplicationsRepo;
+    const sfasApplicationLegacyAwards = sfasApplicationLegacyRepo
+      .createQueryBuilder(`${sfasApplicationType}`)
+      .select(`SUM(${sfasApplicationType}.csgp_award)`, "CSGP")
+      .addSelect(`SUM(${sfasApplicationType}.sbsd_award)`, "SBSD")
+      .addSelect(`SUM(${sfasApplicationType}.csgd_award)`, "CSGD")
+      .addSelect(`SUM(${sfasApplicationType}.bcag_award)`, "BCAG");
+    if (sfasApplicationType === "sfasPTApplication")
+      sfasApplicationLegacyAwards.addSelect(
+        `SUM(${sfasApplicationType}.cspt_award)`,
+        "CSPT",
+      );
+    sfasApplicationLegacyAwards
+      .innerJoin(`${sfasApplicationType}.individual`, "sfasStudent")
+      .where("lower(sfasStudent.lastName) = lower(:lastName)", { lastName })
+      .andWhere("sfasStudent.sin = :sin", { sin })
+      .andWhere("sfasStudent.birthDate = :birthDate", { birthDate })
```

Comment:
@andrewsignori-aot @andrepestana-aot @guru-aot @sh16011993 

Not a part of the PR. 

During student profile creation(internalCreateStudent) if we update the studentId to `sfas_individual` table when a match is found, then we do not have to do this match everytime for legacy data lookup right??

Reply:
If the student becomes a part of the SIMS system first and then gets registered to the SFAS system (not sure if this is even possible), then we would need this check every time, right? Please let me know if I am missing something.

Reply:
A student coming from SFAS bridge will have their IDs updated upon data import @sh16011993.
What @dheepak-aot was saying is that we can ensure the id association during student creation also and start to rely more on the student id from SFAS tables.